core/notifications/src/notification_event/mod.rs

```diff
@@ -58,8 +61,11 @@ pub trait NotificationEvent: std::fmt::Debug + Send + Sync {
     fn should_be_added_to_history(&self) -> bool {
         false
     }
-    fn to_localized_persistent_message(&self, _locale: GaloyLocale) -> LocalizedStatefulMessage {
-        unimplemented!()
+    fn to_localized_persistent_message(
+        &self,
+        _locale: GaloyLocale,
+    ) -> Option<LocalizedStatefulMessage> {
+        None
```

Comment:
This is the wrong way. The interface should encapsulate the contract of the should method is true we  always return the message. Anything else is an implementation bug that is ok to crash (since it should never happen). 

So if you refactor remove the other options instead of adding this one.

Reply:
Makes sense, I will make the change